### ISSUE TITLE
stacktrace and crash related to toolchains

### ISSUE THREAD

**Issue by mikedanese** [bazelbuild/rules_go#906]
rules_go at head
$ Build label: 0.6.1
Build target: bazel-out/local-fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Thu Oct 5 21:54:59 2017 (1507240499)
Build timestamp: 1507240499
Build timestamp as int: 1507240499

```
Unhandled exception thrown during build; message: CONFIGURED_TARGET://cmd/identity-apiserver:go_default_library 12a06b13ab6c6e3848bc8bd0617cd285 (1748189262 1891192726) -> <ErrorInfo exception=com.google.devtools.build.lib.skyframe.ToolchainResolutionFunction$NoToolchainFoundException: no matching toolchain found for 
@io_bazel_rules_go//proto:proto rootCauses={ToolchainResolutionKey{configuration=12a06b13ab6c6e3848bc8bd0617cd285, toolchainType=@io_bazel_rules_go//proto:proto, targetPlatform=struct(constraints = [struct(constraint = struct(label = Label("@bazel_tools//platforms:os")), label = Label("@bazel_tools//platforms:linux"))
, struct(constraint = struct(label = Label("@bazel_tools//platforms:cpu")), label = Label("@bazel_tools//platforms:x86_64"))], label = Label("@bazel_tools//platforms:target_platform")), execPlatform=struct(constraints = [struct(constraint = struct(label = Label("@bazel_tools//platforms:os")), label = Label("@bazel_too
ls//platforms:linux")), struct(constraint = struct(label = Label("@bazel_tools//platforms:cpu")), label = Label("@bazel_tools//platforms:x86_64"))], label = Label("@bazel_tools//platforms:host_platform"))}} cycles=[]>
INFO: Elapsed time: 0.407s
java.lang.IllegalStateException: CONFIGURED_TARGET://cmd/identity-apiserver:go_default_library 12a06b13ab6c6e3848bc8bd0617cd285 (1748189262 1891192726) -> <ErrorInfo exception=com.google.devtools.build.lib.skyframe.ToolchainResolutionFunction$NoToolchainFoundException: no matching toolchain found for @io_bazel_rules_g
o//proto:proto rootCauses={ToolchainResolutionKey{configuration=12a06b13ab6c6e3848bc8bd0617cd285, toolchainType=@io_bazel_rules_go//proto:proto, targetPlatform=struct(constraints = [struct(constraint = struct(label = Label("@bazel_tools//platforms:os")), label = Label("@bazel_tools//platforms:linux")), struct(constrai
nt = struct(label = Label("@bazel_tools//platforms:cpu")), label = Label("@bazel_tools//platforms:x86_64"))], label = Label("@bazel_tools//platforms:target_platform")), execPlatform=struct(constraints = [struct(constraint = struct(label = Label("@bazel_tools//platforms:os")), label = Label("@bazel_tools//platforms:lin
ux")), struct(constraint = struct(label = Label("@bazel_tools//platforms:cpu")), label = Label("@bazel_tools//platforms:x86_64"))], label = Label("@bazel_tools//platforms:host_platform"))}} cycles=[]>
        at com.google.common.base.Preconditions.checkState(Preconditions.java:721)
        at com.google.devtools.build.lib.util.Preconditions.checkState(Preconditions.java:238)
        at com.google.devtools.build.lib.skyframe.SkyframeBuildView.assertSaneAnalysisError(SkyframeBuildView.java:408)
        at com.google.devtools.build.lib.skyframe.SkyframeBuildView.configureTargets(SkyframeBuildView.java:278)
        at com.google.devtools.build.lib.analysis.BuildView.update(BuildView.java:600)
        at com.google.devtools.build.lib.buildtool.BuildTool.runAnalysisPhase(BuildTool.java:503)
        at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:215)
        at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:339)
        at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:67)
        at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:600)
        at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:351)
        at com.google.devtools.build.lib.runtime.CommandExecutor.exec(CommandExecutor.java:58)
        at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:850)
        at com.google.devtools.build.lib.server.GrpcServerImpl.access$2100(GrpcServerImpl.java:109)
        at com.google.devtools.build.lib.server.GrpcServerImpl$2.lambda$run$0(GrpcServerImpl.java:915)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
java.lang.IllegalStateException: CONFIGURED_TARGET://cmd/identity-apiserver:go_default_library 12a06b13ab6c6e3848bc8bd0617cd285 (1748189262 1891192726) -> <ErrorInfo exception=com.google.devtools.build.lib.skyframe.ToolchainResolutionFunction$NoToolchainFoundException: no matching toolchain found for @io_bazel_rules_g
o//proto:proto rootCauses={ToolchainResolutionKey{configuration=12a06b13ab6c6e3848bc8bd0617cd285, toolchainType=@io_bazel_rules_go//proto:proto, targetPlatform=struct(constraints = [struct(constraint = struct(label = Label("@bazel_tools//platforms:os")), label = Label("@bazel_tools//platforms:linux")), struct(constrai
nt = struct(label = Label("@bazel_tools//platforms:cpu")), label = Label("@bazel_tools//platforms:x86_64"))], label = Label("@bazel_tools//platforms:target_platform")), execPlatform=struct(constraints = [struct(constraint = struct(label = Label("@bazel_tools//platforms:os")), label = Label("@bazel_tools//platforms:lin
ux")), struct(constraint = struct(label = Label("@bazel_tools//platforms:cpu")), label = Label("@bazel_tools//platforms:x86_64"))], label = Label("@bazel_tools//platforms:host_platform"))}} cycles=[]>
        at com.google.common.base.Preconditions.checkState(Preconditions.java:721)
        at com.google.devtools.build.lib.util.Preconditions.checkState(Preconditions.java:238)
        at com.google.devtools.build.lib.skyframe.SkyframeBuildView.assertSaneAnalysisError(SkyframeBuildView.java:408)
        at com.google.devtools.build.lib.skyframe.SkyframeBuildView.configureTargets(SkyframeBuildView.java:278)
        at com.google.devtools.build.lib.analysis.BuildView.update(BuildView.java:600)
        at com.google.devtools.build.lib.buildtool.BuildTool.runAnalysisPhase(BuildTool.java:503)
        at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:215)
        at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:339)
        at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:67)
        at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:600)
        at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:351)
        at com.google.devtools.build.lib.runtime.CommandExecutor.exec(CommandExecutor.java:58)
        at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:850)
        at com.google.devtools.build.lib.server.GrpcServerImpl.access$2100(GrpcServerImpl.java:109)
        at com.google.devtools.build.lib.server.GrpcServerImpl$2.lambda$run$0(GrpcServerImpl.java:915)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by ixdy** [comment]
@mikedanese I think you need to add
```
load("@io_bazel_rules_go//proto:def.bzl", "proto_register_toolchains")
proto_register_toolchains()
```
to the WORKSPACE, per the comment in the top-level README.

**Comment by mikedanese** [comment]
woooo! thanks!

**Comment by jayconrod** [comment]
@ianthehat Do you know if this is getting fixed in Bazel? I knew this was an issue in 0.5.4, but I didn't think it still happened in 0.6.1.

**Comment by jayconrod** [comment]
This can still happen in some situations in 0.7.0. It's being tracked in bazelbuild/bazel#3928.
